File: DiceRoller.lua
local nome = "Rolador3.0 - Lua Version"
local versao = 3
local desc = "Rolador de dados"
math.randomseed(os.time())

local function rolador(word,eol)
	qtd = word[2]--QTD
	lados = word[3]--LADOS
	mod = word[4]--MOD
	nome = word[5]--PERSONAGEM

	erro = false
	erro = (tonumber(qtd) < 1)
	erro = ((tonumber(lados) < 1) or erro)
	erro = ((not nome) or erro)

	if erro then
		hexchat.print("Por favor utilize /rola QTD LADOS MOD NOME_PERSONAGEM sendo que o QTD, LADOS e NOME_PERSONAGEM não podem ser vazios!!!")
	else

		dado = 0
		dados = 0
		frase = "Rolei "..tostring(qtd).."d"..tostring(lados).." com mod("..tostring(mod)..") para "..nome.." e os resultados foram ["
		for n = 1,qtd do
			if (n > 1) then
				frase = frase.." - "
			end
			dado = math.random(1,lados) 
			dados = dados + dado
			if dado < (lados/3) then
				frase = frase.." 0,4 "..tostring(dado).." "
			end
			if dado >= (lados/3) and dado < ((lados/3)*2) then
				frase = frase.." 0,7 "..tostring(dado).." "
			end
			if dado >= ((lados/3)*2) then
				frase = frase.." 0,3 "..tostring(dado).." "
			end
		end
		frase = frase.." + "..tostring(mod).."]"
		dados = dados + mod
		frase = frase.." = "..tostring(dados)..""

		canal = hexchat.get_info("channel")
		comando = "msg "..canal.." "..frase
		hexchat.command(comando)
	end 

	return hexchat.EAT_ALL
end

hexchat.register(nome,versao,desc)
hexchat.hook_command("ROLA",rolador,desc)

File: DiceRoller.py
__module_name__ = "Dice Roller"
__module_version__ = "1.0"
__module_description__ = "Rolador de Dados em Python"

import hexchat 
from random import seed
from random import randint

def rolarDado(nome, qtd, lados, mod):
	seed()
	texto = "Rolei "+str(qtd)+"d"+str(lados)+" com mod("+str(mod)+") para "+str(nome)+" e os resultados foram ["
	dadoAtual = 1
	resultado = 0
	resultadoFinal = 0
	while dadoAtual <= qtd:
		if dadoAtual > 1:
			texto = texto+" - "
		
		resultado = randint(1,lados)
		resultadoFinal = resultadoFinal + resultado		
		if resultado < (lados/3):
			texto = texto+" 0,4 "+str(resultado)+" "

		if resultado >= (lados/3) and resultado < ((lados/3)*2):
			texto = texto+" 0,7 "+str(resultado)+" "

		if resultado >= ((lados/3)*2):
			texto = texto+" 0,3 "+str(resultado)+" "

		dadoAtual = dadoAtual + 1
	texto = texto+" + "+str(mod)+"]"
	resultadoFinal = resultadoFinal + mod
	texto = texto+" = "+str(resultadoFinal)+""
	return texto


def comandoRolador(word, word_eol, userdata):
	if len(word) < 5:
		print("Por favor utilize /rola QTD LADOS MOD NOME_PERSONAGEM sendo que o QTD, LADOS e NOME_PERSONAGEM não podem ser vazios!!!")
	else:		
		qtd = int(word[1]) #QTD
		lados = int(word[2]) #LADOS
		mod = int(word[3]) #MOD
		nome = word[4] #PERSONAGEM
		frase = rolarDado(nome,qtd,lados,mod)
		canal = hexchat.get_info("channel")
		comando = "msg "+canal+" "+frase
		hexchat.command(comando)
	return hexchat.EAT_ALL


hexchat.hook_command("rola",comandoRolador,"Por favor utilize /rola QTD LADOS MOD NOME_PERSONAGEM sendo que o QTD, LADOS e NOME_PERSONAGEM não podem ser vazios!!!")
